packages/plugins/src/hardhat/deployments.ts

```diff
@@ -164,12 +172,45 @@ export const getContract = async (
       referenceName
     )
   if (contractConfig.kind === 'no-proxy') {
-    const artifact = hre.artifacts.readArtifactSync(contractConfig.contract)
-    address = getContractAddress(
+    const artifactPaths = await getArtifactPaths(
+      hre,
+      userConfig.contracts,
+      hre.config.paths.artifacts,
+      path.join(hre.config.paths.artifacts, 'build-info')
+    )
+
+    await assertValidLibraries(
+      userConfig,
+      artifactPaths,
+      'hardhat',
+      true,
+      process.stderr
+    )
```

Comment:
> Since getContract is called after a config has already been validated and deployed

We can't guarantee that, we have to assume it might be invalid due to user error.

Reply:
Shouldn't we throw an error if the config is invalid due to user error? Before `getContract` is ever called